src/config.c

```diff
@@ -687,63 +632,162 @@ void loadServerConfig(char *filename, char config_from_stdin, char *options) {
     sdsfree(config);
 }
 
+static int performInterfaceSet(standardConfig *config, sds value, const char **errstr) {
+    sds *argv;
+    int argc, res;
+
+    if (config->flags & MULTI_ARG_CONFIG) {
+        argv = sdssplitlen(value, sdslen(value), " ", 1, &argc);
+    } else {
+        argv = (char**)&value;
+        argc = 1;
+    }
+
+    /* Set the config */
+    res = config->interface.set(config->data, argv, argc, errstr);
+    if (config->flags & MULTI_ARG_CONFIG) sdsfreesplitres(argv, argc);
+    return res;
+}
+
+static void restoreBackupConfig(standardConfig **set_configs, sds *old_values, int count, apply_fn *apply_fns) {
+    int i;
+    const char *errstr = NULL;
+    /* Set all backup values */
+    for (i = 0; i < count; i++) {
+        if (!performInterfaceSet(set_configs[i], old_values[i], &errstr))
+            serverPanic("Failed restoring failed CONFIG SET command: %s", errstr);
```

Comment:
Not sure about panicing here.

This should of course never happen, and if it does - it's either
* A bug
* Some external state. For example, you overwrite a TLS cert file and use `CONFIG SET` to force a reload. File is corrupt so the reload fails, but so will the revert.

In both cases, terminating without providing any option to remediate seems a bit harsh and potentially very risky.

Reply:
I agree. I wasn't sure about this either. My approach was to avoid staying in an undefined state, which seems to me the only sane way to handle this other than panicking.
So the other option would be to report an error and just ignore the failure. WDYT?

Reply:
I think this would be better.

Reply:
done.